api/scpca_portal/management/commands/create_portal_metadata.py

```diff
@@ -11,34 +11,45 @@
 logger.setLevel(logging.INFO)
 logger.addHandler(logging.StreamHandler())
 
+# Default values for arguments
+CLEAN_UP_OUTPUT_DATA = settings.PRODUCTION
+UPLOAD_S3 = False
+
 
 class Command(BaseCommand):
     help = """Creates a computed file and zip for portal-wide metadata,
      saves the instance to the databse, and
      uploads the zip file to S3 bucket.
     """
 
-    @staticmethod
-    def clean_up_output_data():
-        """Cleans up the output data files after processing the computed file"""
-        logger.info("Cleaning up output data")
-        # This static method may not be required using buffers
-
     def add_arguments(self, parser):
         parser.add_argument(
-            "--clean-up-output-data", action=BooleanOptionalAction, default=settings.PRODUCTION
+            "--clean-up-output-data", action=BooleanOptionalAction, default=CLEAN_UP_OUTPUT_DATA
         )
+        parser.add_argument("--upload-s3", action=BooleanOptionalAction, default=UPLOAD_S3)
 
     def handle(self, *args, **kwargs):
         self.create_portal_metadata(**kwargs)
 
     def create_portal_metadata(self, **kwargs):
+        clean_up_output_data = kwargs.get("clean_up_output_data", CLEAN_UP_OUTPUT_DATA)
+        upload_s3 = kwargs.get("upload_s3", UPLOAD_S3)
+
         logger.info("Creating the portal-wide metadata computed file")
         computed_file = ComputedFile.get_portal_metadata_file(
             Project.objects.all(), common.GENERATED_PORTAL_METADATA_DOWNLOAD_CONFIG
         )
 
-        if kwargs["clean_up_output_data"]:
-            self.clean_up_output_data()
+        if computed_file:
+            logger.info("Saving the computed file object to the database")
+            computed_file.save()
+
+            if upload_s3:
+                logger.info("Uploading the zip file to S3")
+                computed_file.upload_s3_file()
```

Comment:
This is now out of date. We need to pull in the `scpca_portal.s3` module and call `s3.upload_output_file(computed_file.s3_key, bucket_name=computed_file.s3_bucket)`

Reply:
Replace the method to a new one from the `s3` module  at ffcc45ee90a7635d626843cd4117922b51421a0f.